src/NHPersistenceCompatibilityTests/PersistenceCompatibilityTests/Database.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace PersistenceCompatibilityTests
+{
+    using System.Data.SqlClient;
+    using Common;
+
+    public class Database
+    {
+        public static void Cleanup()
+        {
+            var connectionString = NHibernateConnectionInfo.ConnectionString;
+            var builder = new SqlConnectionStringBuilder(connectionString);
+            var initialCatalog = builder.InitialCatalog;
+            builder.InitialCatalog = "master";
+
+            using (var connection = new SqlConnection(builder.ConnectionString))
+            using (var command = connection.CreateCommand())
+            {
+                connection.Open();
+
+                var query = @"IF EXISTS(SELECT * from sys.databases where name = '{0}')
```

Comment:
This looks like if its TSQL specific. The Dialect property was made public. Does this only run for SQL Server? Shouldn't there be a dialect check?